src/dsp_tools/utils/connection_live.py

```diff
@@ -352,10 +351,9 @@ def _mask(self, sensitive_info: str) -> str:
         else:
             return f"{sensitive_info[:unmasked_until]}[+{len(sensitive_info) - unmasked_until}]"
 
-    def _should_retry(self, response: Response) -> bool:
-        in_500_range = HTTP_SERVER_ERROR_LOWER <= response.status_code <= HTTP_SERVER_ERROR_UPPER
-        try_again_later = "try again later" in response.text
-        return try_again_later or in_500_range
+    def _in_testing_environment(self) -> bool:
+        in_testing_env = os.getenv("DSP_TOOLS_TESTING")  # set in .github/workflows/tests-on-push.yml
+        return in_testing_env == "true"
```

Comment:
if we don't do that, the CI tests run very long, and it takes too much time until we know that a test failed.

In `test_connection_live.py`, we test that calling a non-existing route leads to an error (`self.assertRaises(BaseError, self.con.get, "/doesNotExist")`). DSP-API responds with 404 on the route `/doesNotExist`, and in our tests, we don't want to retry then.